Src/Idioms/ArgumentNullOrEmptyException.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace AutoFixture.Idioms
+{
+    /// <summary>
+    /// Represents an error about a null or empty string.
+    /// </summary>
+    [Serializable]
+    public class ArgumentNullOrEmptyException : ArgumentException
```

Comment:
Sorry for the confusion I caused by my comment, it seems I wasn't clear enough 😖 

We should not introduce this exception in our library. By doing so we'll force people to reference Idioms from the production code - this wrong a bit, as library was designed to be used in unit test project only. Instead, we should expect the `ArgumentException` raised with the correct parameters.

I mentioned `ArgumentNullOrEmptyException` to demonstrate that this kind of exception _might be created by users_ in their production code to reduce the boilerplate code.

Please remove the exception and expect the `ArgumentException` in the expectation you introduced.